### PR TITLE
Add tutorials for Parallel External Source.

### PR DESCRIPTION
**Pull request by klecki**
## Description
- [ ] **Bug fix** (*non-breaking change which fixes an issue*)
- [ ] **New feature** (*non-breaking change which adds functionality*)
- [ ] **Breaking change** (*fix or feature that would cause existing functionality to not work as expected*)
- [ ] **Refactoring** (*Redesign of existing code that doesn't affect functionality*)
- [x] **Other** (*e.g. Documentation, Tests, Configuration*)

### What happened in this PR
Two part tutorial for Parallel external source.
Two workarounds the limitations of Jupyter Notebook:
* writing a callable definition to temporary file, 
    so it can be loaded and serialized.
* clear notebook without CUDA context to use 'fork'

#### Additional information
- Affected modules and functionalities:
Only tutorials

- Key points relevant for the review:
Mostly the wording, the intro might be longish.

### Tests
- [x] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [x] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [x] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: DALI-2215

Signed-off-by: Krzysztof Lecki <klecki@nvidia.com>

### CODE REVIEW

**Comment by klecki** [comment]
!build

**Comment by klecki** [comment]
!build

**Comment by JanuszL** [comment]
Maybe we should rename `ExternalSource Operator` to `External Source - basic usage` or something to have it aligned with new examples.

**Comment by JanuszL** [comment]
> I think there should not be any particular problem with supporting that method too, it just didn't seem to benefit us because forkserver process is spawned, thus arguments passed to worker are pickled and the advantage of inheriting state (and easily allowing lambdas etc) is lost. I believe main advantage is lower overhead of starting larger pools, because: 1. spawn is expensive, 2. if callback needs some expensive to import modules user should be able to specify what modules are available in spawned forkserver process according to [this](https://stackoverflow.com/questions/63424251/multiprocessing-in-python-what-gets-inherited-by-forkserver-process-from-paren) so those imports don't have to be repeated in each of the workers.
> 
> I can investigate further supporting this method, or postpone it until there's somebody with use case where it actually would benefit them.
> 
> [View entire conversation](https://app.reviewnb.com/NVIDIA/DALI/pull/3199/discussion/) on ReviewNB

I think it is enough just to say that in DALI case forkserver  doesn't provide any benefit or difference that is why we didn't implemented its support (as we start pool only once so the perf impact is negligible).

**Comment by klecki** [comment]
> Maybe we should rename `ExternalSource Operator` to `External Source - basic usage` or something to have it aligned with new examples.

Done

**Comment by klecki** [comment]
!build

**Comment by klecki** [comment]
!build

**Review by JanuszL — APPROVED** [review]

**Review by JanuszL — APPROVED** [review]

File: docs/examples/general/data_loading/external_input.ipynb [hunk 1]
```diff
@@ -4,7 +4,7 @@
    "cell_type": "markdown",
```
Comment (stiepan) [inline docs/examples/general/data_loading/external_input.ipynb:4]: <pre class="ql-syntax" spellcheck="false">Line #9.                self.files = [line.rstrip() for line in f if line is not '']
</pre><p>!= ''</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA/DALI/pull/3199/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (klecki) [reply docs/examples/general/data_loading/external_input.ipynb:4]: Done

**Comment by klecki** [comment]
!build

**Review by stiepan — APPROVED** [review]

**Comment by klecki** [comment]
!build

**Pull request merged**